system/lib/libcxxabi/src/cxa_exception_js_utils.cpp

```diff
@@ -78,6 +86,13 @@ void __get_exception_message(void* thrown_object, char** type, char** message) {
     static_cast<const __shim_type_info*>(&typeid(std::exception));
   int can_catch = catch_type->can_catch(thrown_type, thrown_object);
   if (can_catch) {
+#if __WASM_EXCEPTIONS__
+    if (isDependentException(&exception_header->unwindHeader))
+      thrown_object =
+        reinterpret_cast<__cxa_dependent_exception*>(exception_header)
+          ->primaryException;
```

Comment:
I wish I could say I understood the whole exception / dependent exception / unwind header stuff... but if this fixes the issue then lgtm!

Reply:
It looks when you do `std::rethrow_exception`, internally a dependent exception of the original (primary) exception is created and thrown. But that dependent exception apparently cannot be cast to `__cxa_exception` and can't be called `what()` to get the message. So this is just getting back the original primary exception to get the message.